File: 775.global-and-local-inversions.py
#
# @lc app=leetcode id=775 lang=python3
#
# [775] Global and Local Inversions
#

# @lc code=start
class Solution:
    def isIdealPermutation(self, nums: List[int]) -> bool:
        for i, num in enumerate(nums):
            if i - num not in {-1, 0, 1}:
                return False
        return True


# @lc code=end

File: 775.global-and-local-inversions.rs
/*
 * @lc app=leetcode id=775 lang=rust
 *
 * [775] Global and Local Inversions
 */

// @lc code=start
impl Solution {
    pub fn is_ideal_permutation(nums: Vec<i32>) -> bool {
        nums.into_iter()
            .enumerate()
            .all(|(i, v)| matches!(v - i as i32, 0 | 1 | -1))
    }
}
// @lc code=end
